tests/models/gpt2/test_modeling_flax_gpt2.py

```diff
@@ -335,3 +335,42 @@ def test_model_from_pretrained(self):
             model = model_class_name.from_pretrained("gpt2", from_pt=True)
             outputs = model(np.ones((1, 1)))
             self.assertIsNotNone(outputs)
+
+    def test_bool_attention_mask_in_generation(self):
```

Comment:
Could we make this a fast test? We can first define a `check_bool_attention_mask_in_generation ` function in the model tester:
```python
    def check_bool_attention_mask_in_generation(self, model_class_name, config, input_ids, attention_mask):
        model = model_class_name(config)
        attention_mask_bool = input_ids != 0

        output_bool_att_mask = model.generate(
            input_ids=input_ids,
            attention_mask=attention_mask_bool,
            max_new_tokens=3,
        )

        output_int_att_mask = model.generate(
            input_ids=input_ids,
            attention_mask=attention_mask_int,
            max_new_tokens=3,
        )

        self.assertListEqual(output_bool_att_mask.sequences,  output_int_att_mask.sequences, "Generated response differ between boolean and integer attention mask",)
```

And then a test function in the model test:
```python
    def test_bool_attention_mask_in_generation(self):
        for model_class_name in all_generative_model_classes:
             config, input_ids, attention_mask = self.model_tester.prepare_config_and_inputs()
            self.model_tester. check_bool_attention_mask_in_generation(model_class_name, config, input_ids, attention_mask)
```